analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,296 @@
+                        packages.add(createPackageFromLocalRepository(id, revision))
+                    } else {
+                        packages.add(createPackageFromGenericGitUrl(id, revision))
+                    }
+                }
+                DependencyType.BINARY -> {
+                    // ID is an URL or a path to a file that contains a Carthage binary project specification.
+                    if (isFilePath(workingDir, id)) {
+                        val filePath = id.removePrefix("file://")
+                        val binarySpecFile = when {
+                            isAbsolutePath(filePath) -> File(filePath)
+                            else -> File("$workingDir/$filePath")
+                        }
+                        val content = binarySpecFile.readText()
+                        val binarySpec = jsonMapper.readValue<Map<String, String>>(content)
+                        packages.add(createPackageFromBinarySpec(binarySpec, id, revision))
+                    } else {
+                        val content = URL(id).readText()
+                        val binarySpec = jsonMapper.readValue<Map<String, String>>(content)
+                        packages.add(createPackageFromBinarySpec(binarySpec, id, revision))
+                    }
+                }
+            }
+        }
+
+        return packages
+    }
+
+    private fun parseDependencyLine(line: String): DependencyDefinition {
+        val split = line.split(" ", limit = 3)
+        val type = DependencyType.valueOf(split[0].toUpperCase())
+        val id = split[1].removeSurrounding("\"")
+        val revision = split[2].removeSurrounding("\"")
+
+        return DependencyDefinition(type, id, revision)
+    }
+
+    private fun createPackageFromGenericGitUrl(id: String, revision: String): Package {
+        val vcsInfoFromUrl = VcsHost.toVcsInfo(id)
+        val vcsInfo = VcsInfo(
+            type = vcsInfoFromUrl.type,
+            url = vcsInfoFromUrl.url,
+            revision = revision
+        )
+        val vcsHost = VcsHost.fromUri(URI(vcsInfo.url))
+        val projectUrl = id.removeSuffix(".git")
+
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = vcsHost?.getUserOrOrganization(projectUrl).orEmpty(),
+                name = vcsHost?.getProject(projectUrl).orEmpty(),
+                version = revision
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = projectUrl,
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact(
+                url = vcsHost?.getArchiveDownloadUrl(vcsInfo).orEmpty(),
+                hash = Hash.NONE
+            ),
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromGithubUserAndProject(id: String, revision: String): Package {
```

Comment:
Why did you choose to add `Github-specific` code? Is it only for setting the source artifact location?

Reply:
Carthage supports `github` dependency types which do not require a full git repository URL. Instead they just require `username/project`.
That's why this implementation is github specific. To my knowledge Carthage only allows this syntax for Github.
See: https://github.com/Carthage/Carthage/blob/master/Documentation/Artifacts.md#example-cartfile

However, I changed the signature of this method to be more precise:
`private fun createPackageFromGithubUserAndProject(projectInfo: ProjectInfo): Package`

Reply:
BTW, I believe this is similar to the Node-specific GitHub shortcuts we support for NPM at

https://github.com/oss-review-toolkit/ort/blob/621830f67a420519b9e1808731668dcab5105f28/analyzer/src/main/kotlin/managers/utils/NodeSupport.kt#L75-L111

Reply:
Thanks! I see that there are lines in the example starting with `github` which thus indeed require github specific handling.

This leads me to the question whether it makes sense to reduce the amount of functions which create a `Package` instance would make sense. That is, if we observe a line starting with `github`, we could parse that into a `VcsInfo` or `VCS URL` and then call a function which creates a package for that given `VcsInfo`. The same would apply the lines starting with just `git`. I thought that'd make the package creation logic a bit less redundant. Does that make any sense?

Reply:
`github` and `git` dependencies both call now the private `createPackageFromGenericGitUrl` method. No need to separate them.